File: client-java/controller/src/main/java/org/evomaster/client/java/controller/mongo/selectors/GreaterThanSelector.java
package org.evomaster.client.java.controller.mongo.selectors;

import org.evomaster.client.java.controller.mongo.operations.GreaterThanOperation;
import org.evomaster.client.java.controller.mongo.operations.QueryOperation;

import java.util.Objects;

/**
 * { field: { $gt: value } }
 */
public class GreaterThanSelector extends SingleConditionQuerySelector {

    public static final String GREATER_THAN_OPERATOR = "$gt";

    @Override
    protected QueryOperation parseValue(String fieldName, Object value) {
        Objects.requireNonNull(fieldName);

        if (value == null) {
            return null;
        } else {
            return new GreaterThanOperation<>(fieldName, value);
        }
    }

    @Override
    protected String operator() {
        return GREATER_THAN_OPERATOR;
    }
}

File: client-java/controller/src/main/java/org/evomaster/client/java/controller/mongo/selectors/GreaterThanSelector.kt
package org.evomaster.client.java.controller.mongo.selectors

import org.evomaster.client.java.controller.mongo.operations.GreaterThanOperation
import org.evomaster.client.java.controller.mongo.operations.QueryOperation

/**
 * { field: { $gt: value } }
 */
class GreaterThanSelector : SingleConditionQuerySelector() {
    override fun operator(): String {
        return "\$gt"
    }

    override fun parseValue(fieldName: String, value: Any?): QueryOperation = GreaterThanOperation(fieldName,value)
}